src/axom/klee/GeometryOperatorsIO.cpp

```diff
@@ -0,0 +1,662 @@
+                     const TransformableGeometryProperties &startProperties)
+{
+  verifyObjectFields(opContainer, "translate", FieldSet {}, FieldSet {});
+  return std::make_shared<Translation>(
+    toVector(opContainer, "translate", startProperties.dimensions),
+    startProperties);
+}
+
+/**
+ * Parse a "rotate" operator.
+ *
+ * \param opContainer the Container from which to read the operator
+ * \param startProperties the properties prior to this operator
+ * \return the created operator
+ */
+OpPtr parseRotate(const inlet::Container &opContainer,
+                  const TransformableGeometryProperties &startProperties)
+{
+  if(startProperties.dimensions == Dimensions::Two)
+  {
+    verifyObjectFields(opContainer, "rotate", FieldSet {}, {"center"});
+    Vector3D axis {0, 0, 1};
+    return std::make_shared<Rotation>(
+      opContainer["rotate"].get<double>(),
+      toPoint(opContainer, "center", Dimensions::Two, Point3D {0, 0, 0}),
+      axis,
+      startProperties);
+  }
+  else
+  {
+    verifyObjectFields(opContainer, "rotate", {"axis"}, {"center"});
+    return std::make_shared<Rotation>(
+      opContainer["rotate"].get<double>(),
+      toPoint(opContainer, "center", Dimensions::Three, Point3D {0, 0, 0}),
+      toVector(opContainer, "axis", Dimensions::Three),
+      startProperties);
+  }
+}
+
+/**
+ * Make a slice, ensuring all the values are valid.
+ *
+ * \param origin the origin of the coordinate system
+ * \param normal a vector normal to the plane
+ * \param up a vector which defines the positive Y direction
+ * \param startProperties the properties before the slice
+ * \param path the path where the slice is specified, for error reporting
+ * \return the created operator
+ * \throws KleeError if any value is invalid
+ */
+OpPtr makeCheckedSlice(Point3D origin,
+                       Vector3D normal,
+                       Vector3D up,
+                       const TransformableGeometryProperties &startProperties,
+                       const Path &path)
+{
+  if(normal.is_zero())
+  {
+    throw KleeError({path, "The 'normal' vector must not be a zero vector"});
+  }
+  if(!utilities::isNearlyEqual(normal.dot(up), 0.0))
+  {
+    throw KleeError(
+      {path, "The 'normal' and 'up' vectors must be perpendicular"});
+  }
+  return std::make_shared<SliceOperator>(origin, normal, up, startProperties);
+}
+
+/**
+ * Get the origin to use for a perpendicular slice.
+ *
+ * \param sliceProxy the Proxy describing the slice
+ * \param planeName the name of the plane ("x", "y", or "z")
+ * \param defaultNormal the default normal vector
+ * \return the point to use as the origin
+ */
+primal::Point3D getPerpendicularSliceOrigin(const inlet::Proxy &sliceProxy,
+                                            char const *planeName,
+                                            const primal::Vector3D &defaultNormal)
+{
+  double axisIntercept = sliceProxy[planeName].get<double>();
```

Comment:
Minor: I believe since `double` is considered "primitive" in Inlet this can be abbreviated to
```cpp
double axisIntercept = sliceProxy[planeName];
```